lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -656,6 +659,55 @@ sub clear_context {
     $adapter_context= undef;
 }
 
+=head2 mask_sensitive
+
+Mask sensitive data in the message and logs error in case of failure
+
+=over 4
+
+=item * C<$message> string - The message to be masked
+
+=back
+
+Returns string - The masked message
+
+=cut
+
+sub mask_sensitive {
+    my ($message) = @_;
+
+    # Define a lookup hash reference for all sensitive data regex patterns to be logged
+    my $sensitive_patterns = {
+        'PII' => [
+            qr/\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b/,  # Email
+        ],
+        'Sensitive' => [
+            qr/\b(?:token|key|oauth[ _-]?token)\s*[:=]\s*([^\s]+)/i, # token or api key
+        ],
+        'Token' => [
+            qr/(?:a1|r1|ct1)-[a-zA-Z0-9]{29}/,  # OAuth, Refresh, and CTrader token patterns
+            qr/[a-zA-Z0-9]{15}/,                  # API Token pattern
+        ],
+    };
+
+    try {
+        foreach my $pattern_ref (map { @$_ } values %$sensitive_patterns) {
+            $message =~ s/$pattern_ref/'*' x length($&)/ige;
+        }
```

Comment:
Following our discussion, maybe it indeed makes sense, not to have the hashref but arrayref with comments, describing the regexes? That way looping over patterns can be simplified.
```suggestion
    my $sensitive_patterns = [
        qr/\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b/,  # Email
        qr/\b(?:token|key|oauth[ _-]?token)\s*[:=]\s*([^\s]+)/i,           # token or api key
        qr/(?:a1|r1|ct1)-[a-zA-Z0-9]{29}/,                                              # OAuth, Refresh, and CTrader token patterns
        qr/[a-zA-Z0-9]{15}/,                                                                    # API Token pattern
    ];

    try {
        foreach my $pattern ($sensitive_patterns->@*) {
            $message =~ s/$pattern/'*' x length($pattern)/ige;
        }
```

Reply:
A list even more optimized, we can drop the dereferencing.

Reply:
Hash lookup are faster and it also makes code more readable and maintainable, if we convert this to list we will rule out other functionalities which might be move achievable in hash data structure. For now we are only looping through each so using any wont make much difference 
please confirm @hani-deriv or @denissafiullin-deriv which approach to use and I will change the implementation to it but I dont see a good reason to change this

Reply:
I agree, having a hash ref makes it more categorized, the thing is here we are dereferencing this hash on each log entry. So basically no real benefit of the hash lookup. at the end what we want is to loop on the values and apply the patterns to the message. If we would maintain the hash, I'd suggest calculating the list once out of it and use it latter on.
Though I've tested the difference with 10k logs each 100 chars length, the list is faster by 10-20ms. which is not much of a performance gain.

Reply:
@hani-deriv @denissafiullin-deriv this is converted into list